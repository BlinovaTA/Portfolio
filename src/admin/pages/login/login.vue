<template lang="pug">
  .login-page-component
    .content
      form.form(@submit.prevent="handleSubmit")
        .form-title Авторизация
        .row
          app-input(
            title="Логин"
            icon="user"
            v-model="user.name"
            :errorMessage="validation.firstError('user.name')"
          )
        .row
          app-input(
            title="Пароль"
            icon="key"
            type="password"
            v-model="user.password"
            :errorMessage="validation.firstError('user.password')"
          )
        .btn
          app-button(title="Отправить")
</template>

<script>
import appInput from "../../components/input";
import appButton from "../../components/button";
import { Validator, mixin as ValidatorMixin } from "simple-vue-validator";

export default {
  mixins: [ValidatorMixin],
  validators: {
    "user.name": value => Validator.value(value).required("Введите имя пользователя"),
    "user.password": value => Validator.value(value).required("Введите пароль")
  },
  components: {
    appInput,
    appButton
  },
  data() {
    return {
      user: {
        name: "",
        password: ""
      }
    }
  },
  methods: {
    handleSubmit() {
      this.$validate().then(isValid => {
        if (isValid) return;

        conso;e.log("request");
      })
    }
  }
}
</script>

<style lang="postcss" scoped src="./login.pcss"></style>