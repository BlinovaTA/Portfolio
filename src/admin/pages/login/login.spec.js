import login from "./index";
import { shallowMount, mount } from "@vue/test-utils";
import { afterEach } from "jest-circus";

let wrapper;

beforeEach(() => {
  wrapper = shallowMount(login);
});

afterEach(() => {
  wrapper.destroy();
});

describe("Login tests", () => {
  it("there is a send button", () => {
    const buttonSubmit = wrapper.find("button");
    expect(buttonSubmit).toBeDefined();
  });

  it("there is a login field", () => {
    const textInputLogin = wrapper.find('input[type="text"]');
    expect(textInputLogin).toBeDefined();
  });

  it("there is a password field", () => {
    const textInputPassword = wrapper.find('input[type="password"]');
    expect(textInputPassword).toBeDefined();
  });

  it("login snapshot", async () => {
    const mountWrapper = mount(login);

    expect(mountWrapper.element).toMatchSnapshot();
  });
});
