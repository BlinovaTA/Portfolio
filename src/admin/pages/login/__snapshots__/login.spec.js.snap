// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests login snapshot 1`] = `
<div
  class="login-page-component"
>
  <div
    class="content"
  >
    <form
      class="form"
    >
      <div
        class="form-title"
      >
        Авторизация
      </div>
      <div
        class="row"
      >
        <label
          class="input input_labeled  input_iconed input_icon-user"
        >
          <div
            class="title"
          >
            Логин
          </div>
           
          <input
            class="input__elem field__elem"
          />
           
          <div
            class="input__error-tooltip"
          >
            <div
              class="input__error-tooltip-container"
            >
              
            </div>
          </div>
        </label>
      </div>
      <div
        class="row"
      >
        <label
          class="input input_labeled  input_iconed input_icon-key"
        >
          <div
            class="title"
          >
            Пароль
          </div>
           
          <input
            class="input__elem field__elem"
            type="password"
          />
           
          <div
            class="input__error-tooltip"
          >
            <div
              class="input__error-tooltip-container"
            >
              
            </div>
          </div>
        </label>
      </div>
      <div
        class="btn"
      >
        <button
          class="default-btn-container btn-decorator"
          title="Отправить"
        >
          Отправить
        </button>
      </div>
    </form>
  </div>
</div>
`;
